streams/src/test/java/org/apache/kafka/test/StreamsTestUtils.java

```diff
@@ -78,4 +72,24 @@ public static Properties minimalStreamsConfig() {
         return results;
     }
 
+    // TODO: these three static functions are added because some non-TopologyBuilder unit tests need to access the internal topology builder,
+    //       which is usually a bad sign of design patterns between TopologyBuilder and StreamThread. We need to consider getting rid of them later
+    public static InternalTopologyBuilder internalTopologyBuilder(final StreamsBuilder streamsBuilder) throws NoSuchFieldException, IllegalAccessException {
```

Comment:
@ijuma that is possible. We'd need a couple of test helper classes in the appropriate packages, one for `StreamBuilder` and one for `InternalStreamsBuilder`. We'd also need to make `internalTopologyBuilder` on `StreamsBuilder` package private.